opennms-webapp-rest/src/main/java/org/opennms/web/rest/v2/EnhanceLinkdRestService.java

```diff
@@ -0,0 +1,236 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2008-2021 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2021 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.web.rest.v2;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+
+import org.opennms.netmgt.dao.api.NodeDao;
+import org.opennms.web.enlinkd.BridgeLinkNode;
+import org.opennms.web.enlinkd.CdpElementNode;
+import org.opennms.web.enlinkd.CdpLinkNode;
+import org.opennms.web.enlinkd.EnLinkdElementFactoryInterface;
+import org.opennms.web.enlinkd.IsisElementNode;
+import org.opennms.web.enlinkd.IsisLinkNode;
+import org.opennms.web.enlinkd.LldpElementNode;
+import org.opennms.web.enlinkd.LldpLinkNode;
+import org.opennms.web.enlinkd.OspfElementNode;
+import org.opennms.web.enlinkd.OspfLinkNode;
+import org.opennms.web.rest.v2.api.EnhanceLinkdRestApi;
+import org.opennms.web.rest.v2.models.BridgeLinkNodeDTO;
+import org.opennms.web.rest.v2.models.CdpElementNodeDTO;
+import org.opennms.web.rest.v2.models.CdpLinkNodeDTO;
+import org.opennms.web.rest.v2.models.IsisElementNodeDTO;
+import org.opennms.web.rest.v2.models.IsisLinkNodeDTO;
+import org.opennms.web.rest.v2.models.LldpElementNodeDTO;
+import org.opennms.web.rest.v2.models.LldpLinkNodeDTO;
+import org.opennms.web.rest.v2.models.OspfElementNodeDTO;
+import org.opennms.web.rest.v2.models.OspfLinkNodeDTO;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+@Component
+@Transactional
+public class EnhanceLinkdRestService implements EnhanceLinkdRestApi {
+
+    private EnLinkdElementFactoryInterface enLinkdElementFactory;
```

Comment:
Have @Autowired on the member variables as per our  current convention

Reply:
I have no problem changing that. Just want to do one more confirmation with you since field injection is normally not recommended in Spring. But I'll respect the project convention if we decide to do that.

Reply:
Yeah, constructor based injection may be better but here we are still doing setter based injection. Don't think that's any better than field based. 
I may have some bias on having @Autowired on fields because that's what I see everywhere in OpenNMS code.

Reply:
Okay, code changed.